src/storage/src/hummock/local_version_manager.rs

```diff
@@ -511,21 +508,58 @@ impl LocalVersionManager {
         for result in join_all(join_handles).await {
             result.expect("should be able to join the flush handle")
         }
-        let (order_index, task_payload, task_write_batch_size) = match self
-            .local_version
-            .write()
-            .get_mut_shared_buffer(epoch)
-            .and_then(|shared_buffer| shared_buffer.new_upload_task(SyncEpoch))
-        {
-            Some(task) => task,
-            None => {
-                tracing::trace!("sync epoch {} has no more task to do", epoch);
-                return Ok(0);
+        let notify1 = self.sync_epoch_notify.clone();
+        // TODO(xuxinhao): modify it after supporting uploading multiple shared buffers.#4442
+        loop {
+            let epoch1 = self
+                .local_version
+                .read()
+                .iter_shared_buffer()
+                .find(|(key, _)| key > &&self.max_sync_epoch.read().clone())
```

Comment:
Can we avoid `iter_shared_buffer` here? I think `min_unsynced_epoch ` should be the smallest epoch in shared buffer.